Dfe.Data.SearchPrototype/Web/Models/ViewModels/Shared/Pagination.cs

```diff
@@ -0,0 +1,118 @@
+        /// </summary>
+        public bool IsPageable => TotalNumberOfPages > 1;
+
+        /// <summary>
+        /// Sets the default first page number to 1.
+        /// </summary>
+        public int FirstPageNumber => 1;
+
+        /// <summary>
+        /// Determines whether the current page is the last page in the sequence.
+        /// </summary>
+        //public bool IsLastPage => CurrentPageNumber == TotalNumberOfPages;
+
+        public bool IsLastPage => NextPageNumber == null;
+
+        /// <summary>
+        /// Determines whether the current page is the first page in the sequence.
+        /// </summary>
+        public bool IsFirstPage => PreviousPageNumber == null;
+
+        /// <summary>
+        /// Determines the current page sequence (i.e. array of page numbers)
+        /// based on the current page and the total number of pages available.
+        /// </summary>
+        public int[] CurrentPageSequence =>
+            _pager.GetPageSequence(CurrentPageNumber, TotalNumberOfPages);
+
+        /// <summary>
+        /// Determines whether the current page falls within the lower paging
+        /// boundary, given the total number of pages.
+        /// </summary>
+        public bool PageSequenceIncludesFirstPage =>
+            _pager.PageSequenceIncludesFirstPage(CurrentPageNumber, TotalNumberOfPages);
+
+        /// <summary>
+        /// Determines whether the current page falls within the lower paging threshold.
+        /// </summary>
+        public bool HasMoreLowerPagesAvailable =>
+            _pager.HasMoreLowerPagesAvailable(CurrentPageNumber, TotalNumberOfPages);
+
+        /// <summary>
+        /// Determines whether the current page falls within the upper paging
+        /// boundary, given the total number of pages provisioned.
+        /// </summary>
+        public bool PageSequenceIncludesLastPage =>
+            _pager.PageSequenceIncludesLastPage(CurrentPageNumber, TotalNumberOfPages);
+
+        /// <summary>
+        /// Determines whether the current page falls within the upper paging threshold.
+        /// </summary>
+        public bool HasMoreUpperPagesAvailable =>
+            _pager.HasMoreUpperPagesAvailable(CurrentPageNumber, TotalNumberOfPages);
+
+        /// <summary>
+        /// Gets the total number of pages.
+        /// </summary>
+        /// <returns>
+        /// The calculated number of pages available.
+        /// </returns>
+        /// <exception cref="ArgumentException">
+        /// Thrown if either we have a total record count of zero, or records per-page
+        /// configured to zero.
+        /// </exception>
+        private int GetTotalNumberOfPages()
+        {
+            if (TotalRecordCount == 0) {
+                throw new ArgumentException("The record count must be greater than zero.");
+            }
+
+            if (RecordsPerPage == 0){
+                throw new ArgumentException("The page size must be greater than zero.");
+            }
```

Comment:
if we need to throw an exception about this, it's probably too late to do it here